Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift

```diff
@@ -843,14 +843,26 @@ extension ProjectDescription.ResourceFileElements {
         // Add default resources path if necessary
         // They are handled like a `.process` rule
         if sources == nil {
-            resourceFileElements += try defaultResourcePaths(from: path)
+            // Already included resources should not be added as default resource
+            let excludedPaths: Set<AbsolutePath> = Set(
+                resourceFileElements.map {
+                    switch $0 {
+                    case let .folderReference(path: path, _, _):
+                        AbsolutePath(stringLiteral: path.pathString)
+                    case let .glob(pattern: path, _, _, _):
+                        // TODO: What's the correct way to remove the ** pattern from the string?
+                        AbsolutePath(stringLiteral: path.pathString).parentDirectory
```

Comment:
Correct me if I'm wrong but does `PackageDescription`'s [Resource](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#resource) support globs? I believe the answer is no.

I'd try to add a glob and see if it works – if it does not, we can throw here and say a glob here is unexpected.

Reply:
I definitely run into this with my sample. I don't know, if PackageDescription supports it. But Tuist converts `resources: [.process("SPMResources")]` into what I get here `/Users/kaioelfke/Desktop/Samples/EnsembleSample/SymLinkResPkg/Sources/SPMResources/**`.

So the filter or closure needs to make sure no default resources are included again that match this glob.

The resource file maps are mapped before and for process rules there's this making a glob:

```swift
func handleProcessResource(resourceAbsolutePath: AbsolutePath) throws -> ProjectDescription.ResourceFileElement? {
    let absolutePathGlob = resourceAbsolutePath.extension != nil ? resourceAbsolutePath : resourceAbsolutePath
        .appending(component: "**")
    for exclude in excluding {
        if absolutePathGlob.isDescendantOfOrEqual(to: path.appending(try RelativePath(validating: exclude))) {
            return nil
        }
    }
    return .glob(
        pattern: .path(absolutePathGlob.pathString),
        excluding: try excluding.map {
            let excludePath = path.appending(try RelativePath(validating: $0))
            let excludeGlob = excludePath.extension != nil ? excludePath : excludePath.appending(component: "**")
            return .path(excludeGlob.pathString)
        }
    )
}
```

Reply:
Gotcha, thanks for the explanation 👍